Update platform UI with some minor improvements

frozenhelium opened the pull request:
- Add browse by Category and Grade in footer
- Fix book list on state change

frozenhelium committed:
Update platform UI with some minor improvements

- Add browse by Category and Grade in footer
- Fix book list on state change

tnagorra commented on app/Base/configs/lang.ts:
```diff
@@ -390,6 +390,14 @@ export const footer = {
         en: 'Contact Us',
         ne: 'सम्पर्क गर्नुहोस',
     },
+    exploreByGradeHeading: {
+        en: 'Grade',
+        ne: 'Grade',
+    },
+    exploreByCategoryHeading: {
+        en: 'Category',
+        ne: 'Category',
+    },
```
Let's translate this strings

tnagorra commented on app/views/BookList/index.tsx:
```diff
@@ -187,10 +188,27 @@ function Explore(props: Props) {
 
+    useDidUpdateEffect(() => {
+        if (locationState) {
+            if (isDefined(locationState.category)) {
+                setCategories([locationState.category]);
+            }
+
+            if (isDefined(locationState.grade)) {
+                setGrade(locationState.grade);
+            }
+
+            if (isDefined(locationState.publisher)) {
+                setPublisher(locationState.publisher);
+            }
```
Let's clear everything except what is defined

tnagorra committed:
Add permission checks on Profile page

- Reset value if cart quantity cannot be set
- Reset filters on books page when location.state changes
